### PR TITLE
ClickUp integration docs

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> ClickUp integration docs.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+| Phone | Must be a valid phone number with a country code, for example, `+1 123 456 7890`. |
+| Date | Must be a valid date in a datetime format, for example, `2022-02-25T13:39:21.032Z`. |
+| Text | Must be a string. |
+| Checkbox | Must be in a Boolean format. |
+| Number | - |
+| Currency | Must be a number |
+| Emoji (integer) | Used for rating. The value must be in a range defined in the ClickUp dashboard. |
+| Location | Latitude/longitude and the formatted address must be passed as the [Google Geocoding API](https://developers.google.com/maps/documentation/geocoding/overview?hl=en). |
+| Dropdown | RudderStack will use the dropdown option name to find the UUID using the [`field`](https://api.clickup.com/api/v2/list/list_id/field) endpoint. |
+| Label | RudderStack will the label name to find the UUID using the [`field`](https://api.clickup.com/api/v2/list/list_id/field) endpoint. |
```
Comment (Gauravudia) [inline docs/destinations/streaming-destinations/clickup.mdx:162]: Typo here. `use` is missing.
 RudderStack will `use` the label ...

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+To obtain your ClickUp API token, follow these steps:
+
+1. Log into your [ClickUp dashboard](https://app.clickup.com/).
+2. From the left sidebar, click the **Settings** button at the bottom and go to **MY APPS** > **Apps**.
+3. You can find the API token listed here, as shown:
+
+<img src="../../assets/event-stream-destinations/clickup-api-token.png" alt="ClickUp API token" />
+
+<div class="infoBlock">
+To create a new app with the ClickUp API, refer to this <a href="https://help.clickup.com/hc/en-us/articles/6303422883095-Create-your-own-app-with-the-ClickUp-API">ClickUp Help Guide</a>.
```
Comment (Gauravudia) [inline docs/destinations/streaming-destinations/clickup.mdx:228]: We can remove this. As we are not using the `OAuth` for authentication. 
Currently, We are only supporting authentication with `API token`.

File: docs/destinations/streaming-destinations/clickup.mdx (hunk 1 above)
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/clickup.mdx:162]:
```suggestion
| Label | RudderStack will use the label name to find the UUID using the [`field`](https://api.clickup.com/api/v2/list/list_id/field) endpoint. |
```

File: docs/destinations/streaming-destinations/clickup.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/clickup.mdx]: This note is now removed.

**Review by Gauravudia — APPROVED** [review]

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+- **API Token**: Enter your ClickUp API token.
+
+<div class="infoBlock">
+For more information on obtaining your ClickUp API token, refer to the <Link to="#where-can-i-find-the-clickup-api-token">FAQ</Link> section below.
+</div>
+
+- **List ID**: Enter your ClickUp list ID.
+
+<div class="infoBlock">
+For more information on obtaining your ClickUp list ID, refer to the <Link to="#where-can-i-find-the-clickup-list-id">FAQ</Link> section below.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:48]:
```suggestion
For more information on obtaining your ClickUp API token and list ID, refer to the <Link to="#where-can-i-find-the-clickup-list-id">FAQ</Link> section below.
```

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+## Connection settings
+
+To successfully configure ClickUp as a destination, you will need to configure the following settings:
+
+<img src="../../assets/event-stream-destinations/clickup-connection-settings-1.png" alt="ClickUp connection settings" />
+
+- **API Token**: Enter your ClickUp API token.
+
+<div class="infoBlock">
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:41]: Can remove here and add it in the below note (next suggestion)

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+For more information on using this setting, refer to the <Link to="#adding-custom-fields">Adding custom fields</Link> section below.
+</div>
+
+<div class="infoBlock">
+You can provide multiple ClickUp custom fields associated with different lists. However, note that the custom field name <strong>must be unique</strong> within a list to successfully create a task.
+</div>
+
+- **Client-side Events Filtering**: This setting lets you specify which events should be allowed to flow through to ConvertFlow.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:61]:
```suggestion
- **Client-side Events Filtering**: This setting lets you specify which events should be allowed to flow through to ClickUp.
```

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+<img src="../../assets/event-stream-destinations/clickup-connection-settings-2.png" alt="ClickUp connection settings" />
+
+<div class="warningBlock">
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:65]: Not sure if we should link it to "Client-side Events Filtering" guide as that guide is for device mode supported destinations and ClickUp supports only cloud mode.

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+| Custom Field | Notes |
+| :-----| :-----| 
+| URL | Must be in a valid URL format. |
+| Email | Must be in a valid email format. | 
+| Phone | Must be a valid phone number with a country code, for example, `+1 123 456 7890`. |
+| Date | Must be a valid date in a datetime format, for example, `2022-02-25T13:39:21.032Z`. |
+| Text | Must be a string. |
+| Checkbox | Must be in a Boolean format. |
+| Number | - |
+| Currency | Must be a number |
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:162]:
```suggestion
| Currency | Must be a number. |
```

File: docs/destinations/streaming-destinations/clickup.mdx [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+| Emoji (integer) | Used for rating. The value must be in a range defined in the ClickUp dashboard. |
+| Location | Latitude/longitude and the formatted address must be passed as the [Google Geocoding API](https://developers.google.com/maps/documentation/geocoding/overview?hl=en). |
+| Dropdown | RudderStack will use the dropdown option name to find the UUID using the [`field`](https://api.clickup.com/api/v2/list/list_id/field) endpoint. |
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:165]:
```suggestion
| Dropdown | RudderStack uses the dropdown option name to find the UUID using the [`field`](https://api.clickup.com/api/v2/list/list_id/field) endpoint. |
```

Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/clickup.mdx:165]:
```suggestion
| Dropdown | RudderStack uses the dropdown option name to find the UUID using the [`field`](https://api.clickup.com/api/v2/list/list_id/field) endpoint. |
```

**Review by ishachopra0401 — APPROVED** [review]

File: docs/destinations/streaming-destinations/clickup.mdx (hunk 4 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/clickup.mdx:41]: We're referencing the individual FAQ in each info block. In that case we will need a generic `<Link to="#faq">FAQ</Link>` tag. I'll make the change.

File: docs/destinations/streaming-destinations/clickup.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/clickup.mdx:48]: Need to change the link tag here as discussed above.

File: docs/destinations/streaming-destinations/clickup.mdx (hunk 6 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/clickup.mdx:65]: @ishachopra0401 I was initially conflicted with the same thought, but thought of adding it anyway. we can remove this.

**Pull request merged**